src/superphot_plus/model/classifier.py

```diff
@@ -314,20 +184,23 @@ def run(
             if epoch % 5 == 0:
                 metrics.print_last()
 
+        # Load model with best validation score
         self.load_state_dict(torch.load(model_path))
 
-        labels, names, pred_labels, max_probs = self.test(
-            test_features, test_classes, test_names, test_group_idxs, probs_csv_path
-        )
-
+        # Plot training and validation metrics
         if plot_metrics:
             plot_model_metrics(
                 metrics=metrics.get_values(),
                 num_epochs=num_epochs,
-                plot_name=test_names[0],
+                plot_name=run_id,
                 metrics_dir=metrics_dir,
             )
 
+        if self.test_data is None:
+            return best_valid_loss, valid_acc
```

Comment:
This is returning only a subset of what the docstring says it will return, which may cause confusion for the users.

Reply:
After breaking `run()` into training and testing methods I am now consistently returning the validation loss and accuracy.